packages/relay/src/lib/eth.ts

```diff
@@ -1349,18 +1349,31 @@ export class EthImpl implements Eth {
 
     let result;
     if (address) {
-      result = await this.mirrorNodeClient.getContractResultsLogsByAddress(address, params, undefined, requestId);
+      result = await this.mirrorNodeClient.getContractResultsLogsByAddress(address, params, limitParams, requestId);
     }
     else {
-      result = await this.mirrorNodeClient.getContractResultsLogs(params, undefined, requestId);
+      result = await this.mirrorNodeClient.getContractResultsLogs(params, limitParams, requestId);
     }
 
     if (!result || !result.logs) {
       return [];
     }
 
+    let unproccesedLogs = result.logs;
```

Comment:
Yep, good call. Also was wondering about refactoring and pulling out some logic into separate methods for better readability, because now getLogs is overfilled with if statements. Not necessary in this PR, can be in separate. What do you think ?

Reply:
Was going to say that we should refactor this method, but maybe in a separate PR?

Reply:
Okay, i'm fine with the separate PR if you capture the ticket and note it here :)

Reply:
Done, i've extracted this logic in separate method in the mirror node client. Opened issue for the rest https://github.com/hashgraph/hedera-json-rpc-relay/issues/704